util/collate/collate.go

```diff
@@ -21,10 +21,18 @@ import (
 var (
 	collatorMap         map[string]Collator
 	collatorIDMap       map[int]Collator
+	id2NameMap          map[int32]string
 	newCollationEnabled bool
 	setCollationOnce    sync.Once
 )
 
+// DefaultCollation is default collation of TiDB. If a string column/constant is defined without collation,
+// we use this as its collation.
+const (
+	DefaultCollation = "binary"
```

Comment:
Please notice that the default collation of string in TiDB is not `binary`, it is `utf8mb4_bin`:

https://github.com/pingcap/parser/blob/517beb2e39c20e78cca59b63351aacb5c1ef5845/mysql/charset.go#L587